internal/adminapi/client_test.go

```diff
@@ -0,0 +1,239 @@
+package adminapi
+
+import (
+	"bytes"
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/tls"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"encoding/pem"
+	"fmt"
+	"github.com/stretchr/testify/assert"
+	"io/ioutil"
+	"math/big"
+	"net"
+	"net/http"
+	"net/http/httptest"
+	"os"
+	"strings"
+	"testing"
+	"time"
+)
+
+var ca *x509.Certificate
+var caPEM *bytes.Buffer
+var caPrivateKeyPEM *bytes.Buffer
+var certPEM *bytes.Buffer
+var certPrivateKeyPEM *bytes.Buffer
+
+func TestMakeHTTPClient(t *testing.T) {
+
+	buildTLS(t)
+
+	var opts = HTTPClientOpts{
+		true,
+		"",
+		"",
+		caPEM.String(),
+		nil,
+		"",
+		certPEM.String(),
+		"",
+		certPrivateKeyPEM.String(),
+	}
+
+	httpclient, err := MakeHTTPClient(&opts)
+	if err != nil {
+		t.Errorf("Fail to make the HTTP client - %s", err.Error())
+	}
+
+	assert.NotNil(t, httpclient)
+
+	validate(t, httpclient)
+
+}
+
+func TestMakeHTTPClientWithFilePaths(t *testing.T) {
+
+	buildTLS(t)
+
+	caFile, err := ioutil.TempFile("", "ca.crt")
+	if err != nil {
+		t.Errorf("Fail to create CA certificate file - %s", err.Error())
+	}
+	caFile.Write(caPEM.Bytes())
```

Comment:
we're not checking the error output of `Write` which we might as well do so that we don't fail later in a less clear way:

```suggestion
	writtenBytes, err := caFile.Write(caPEM.Bytes())
	require.NoError(t, err)
	require.Equal(t, caPEM.Len(), writtenBytes)
```